cumulusci/tasks/salesforce/promote_package_version.py

```diff
@@ -302,10 +306,24 @@ def _query_SubscriberPackageVersion(self, spv_id: str) -> Optional[Dict]:
                 "SubscriberPackageId",
             ],
             "SubscriberPackageVersion",
-            where_clause=f"Id='{spv_id}'",
+            where_clause=self._get_spv_where_clause(spv_id),
             raise_error=True,
         )
 
+    def _get_spv_where_clause(self, spv_id: str) -> str:
+        """Get the where clause for a SubscriberPackageVersion query
+
+        Does not include the WHERE.
+        Includes the installation key if provided.
+        """
+        where_clause = f"Id='{spv_id}'"
+
+        if "install_key" in self.options:
```

Comment:
```suggestion
        if self.options.get("install_key"):
```

@zenibako I think there is a tricky edge case here. `"install_key"` may be present in `self.options`, but be `None`. This would happen in the circumstance that a flow containing this task were frozen to be run in MetaDeploy or similar. That'd cause the query to return no results.